bin/heudiconv

```diff
@@ -142,92 +283,133 @@ def find_files(regex, topdir=curdir, exclude=None, exclude_vcs=True, dirs=False)
 find_files.__doc__ %= (_VCS_REGEX,)
 
 
-def process_dicoms(fl, dcmsession=None, basedir=None, dcmfilter=None):
-    """Process list of dicoms and return  seqinfo and file group
+def group_dicoms_into_seqinfos(fl, flfilter=None, dcmfilter=None, grouping='studyUID'):
+    """Process list of dicoms and return seqinfo and file group
+
+    `seqinfo` contains per-sequence extract of fields from DICOMs which
+    will be later provided into heuristics to decide on filenames
 
     Parameters
     ----------
     fl : list of str
       List of files to consider
-    dcmsession : list of int, optional
-      List of session ids (?) for each file if `fl`
-      corresponds to multiple sessions
-    basedir : str, optional
-      Base directory relative to which filenames are provided in `fl`
+    flfilter : callable, optional
+      Applied to each of fl. Should return True if file needs to be kept,
+      False otherwise. Used to filter fl
     dcmfilter : callable, optional
       If called on dcm_data and returns True, it is used to set
       series_id
+    grouping : str ('studyUID', 'accession_number') or None, optional
+        what to group by: studyUID or accession_number
 
     Returns
     -------
-    seqinfo, filegrp : list of list, dict
+    seqinfo : list of list
       `seqinfo` is a list of info entries per each sequence (some entry 
       there defines a key for `filegrp`)
-      `filegrp` is a dictionary with files groupped per each sequence.
+    filegrp : dict
+      `filegrp` is a dictionary with files groupped per each sequence
     """
+    allowed_groupings = ['studyUID', 'accession_number', None]
+    if grouping not in allowed_groupings:
+        raise ValueError('I do not know how to group by {0}'.format(grouping))
+    per_studyUID = grouping == 'studyUID'
```

Comment:
could be done imho either way.  this centralizes comparison to the string value into a variable.  Happen we decide to refactor anything and complicate the logic to decide e.g. on `per_studyUID` value -- will be just a single point to change instead of hunting for all the comparisons to a string value.  So I would leave it as is for now